chemie/elections/models.py

```diff
@@ -24,244 +25,316 @@
+        return False
 
-    def delete_position(self, positions):
-        if type(positions) is Position:
-            position = positions
-            position.delete_candidates(candidates=position.candidates.all())
-            position.delete()
-            self.save()
-            return
-        elif type(positions) is QuerySet:
-            deletables = positions
-        elif type(positions) is list:
-            ids = [position.id for position in positions]
-            deletables = Position.objects.filter(pk__in=ids)
+    @classmethod
+    def latest_election_is_open(cls):
+        "Checking if the latest election object is currently open"
+        try:
+            election = cls.objects.latest("id")
+            if election.is_open:
+                return True
+        except ObjectDoesNotExist:  # TODO: more exceptions?
+            pass
+        return False
+
+    @classmethod
+    def create_new_election(cls):
+        Election.objects.create(is_open=True)
+        cls.clear_all_checkins()  # setter alle RFID-checkins til False (ingen har møtt opp enda)
+
+    @classmethod
+    def clear_all_checkins(cls):
+        # Get all profiles where eligible_for_voting is set to True and set False
+        Profile.objects.filter(eligible_for_voting=True).update(
+            eligible_for_voting=False
+        )
+
+    @classmethod
+    def get_latest_election(cls):
+        return cls.objects.latest("id")
+
+    @classmethod
+    def is_redirected(cls):
+        """
+        Checks if there is an active voting or the election is open,
+        if not return the redirect function to the correct url
+        """
+        if not cls.latest_election_is_open():
+            return True, redirect("elections:admin_start_election")
         else:
-            raise AttributeError
-        for position in deletables:
-            if position in self.positions.all():
-                position.delete_candidates(candidates=position.candidates.all())
-                position.delete()
-            else:
-                raise ValueError
+            election = cls.get_latest_election()
+            if election.current_position is not None:
+                if election.current_position.is_active:
+                    pk = election.current_position.id
+                    return True, redirect("elections:voting_active", pk=pk)
+            return False, None
+
+    def is_voting_active(self):
+        if self.current_position is not None:
+            if self.current_position.is_active:
+                return True
+        return False
+
+    def add_position(self, form):
+        # lager et nytt position objekt som vi legger inn i election
+        # position field
+        new_position = form.cleaned_data["position_name"]
+        current_positions_in_election = list()
+        for i in self.positions.all():
+            current_positions_in_election.append(i.position_name)
+        if new_position not in current_positions_in_election:
```

Comment:
I stedet for for-løkka over kan du bruke (i henhold til input-forslaget jeg skrev over)
`if position_name not in self.positions.all().values_list("position_name", flat=True)`
Dette vil ta querysettet ditt `self.positions.all()` og gjøre det om til en liste av `position_name` for hvert eneste objekt i det originale querysettet.